src/utils/market-positions-utils.ts

```diff
@@ -322,24 +385,31 @@ export function updateMarketPositionsFromRedemption(
     }
     const market = getMarket(
       conditionalTokenAddress,
-      conditions[0],
+      conditionId,
       collateralToken,
       outcomeSlotCount,
       outcomeIndex,
     );
 
-    let position = getMarketPosition(userAddress, market);
+    let position = getMarketPosition(redeemer, market);
 
     // Redeeming a position is an all or nothing operation so use full balance for calculations
     let numerator = payoutNumerators[outcomeIndex];
     let redemptionValue = position.netQuantity
       .times(numerator)
       .div(payoutDenominator);
 
+    updateUserProfit(
+      redeemer,
+      redemptionValue,
+      collateralScaleDec,
+      event.block.timestamp,
+      conditionId,
+    );
+
```

Comment:
Interesting, if this works for redemptions then it means this loop in the previous branch was what could be multiplying the redemption amount (potentially) https://github.com/Polymarket/polymarket-subgraph/pull/24/files#diff-3cfdece17a043bbbaf39f8c8e1bccfd436239f8584623c3b7be98c2185ba8b04R293 @JonathanAmenechi